src/mmw/js/src/modeling/models.js

```diff
@@ -413,7 +414,16 @@ function _alterModifications(rawModifications) {
         pieces = _.filter(pieces, validShape);
     }
 
-    return pieces;
+    return _.map(pieces, function(piece) { //ensures 'value' is uniform type for each piece
+      var p = piece;
+      if ( typeof p.value =='string') {
+        var v = {};
+        if (p.name === reclass) { v.reclass = p.value; }
```

Comment:
Updated based on both comments. I didn't see your second comment until after I'd pushed the first change, so there are 2 commits.